yeonjae/baekjoon/브루트포스/2798-블랙잭.js

```diff
@@ -0,0 +1,40 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const numTypeArr = (str) =>
+    str.split(" ").map((numString) => Number(numString));
+
+  const [cardNum, dealNum] = numTypeArr(input[0]);
+  const numList = numTypeArr(input[1]);
+
+  let tempSum = 0;
+  let result = 0;
+
+  for (let i = 0; i < cardNum; i++) {
+    for (let j = i + 1; j < cardNum; j++) {
+      for (let k = j + 1; k < cardNum; k++) {
+        tempSum = numList[i] + numList[j] + numList[k];
+        if (tempSum > result && tempSum <= dealNum) {
+          result = tempSum;
+        }
+      }
+    }
+  }
+
+  console.log(result);
+});
```

Comment:
```suggestion
  const convertNumberArr = (str) => str.split(" ").map(Number);
  const [cardNum, dealNum] = convertNumberArr(input[0]);
  const numList = convertNumberArr(input[1]);
  let result = 0;

  for (let i = 0; i < cardNum - 2; i++) {
    for (let j = i + 1; j < cardNum - 1; j++) {
      for (let k = j + 1; k < cardNum; k++) {
        const tempSum = numList[i] + numList[j] + numList[k];
        
        if (tempSum <= dealNum) {
          result = Math.max(result, tempSum);
        }
      }
    }
  }

  console.log(result);
});
```
- 풀이 잘봤습니다 🙇‍♂️
- 입력값을 숫자 배열로 바꿔주는 함수를 따로 만든것은 좋았습니다 그런데 변수명이 함수기 때문에 접두사로 동사 형태로 가져가는 것이 좋습니다 ~ !
- 또한, tempSum은 3번째 for문에서만 사용하는 것이기 때문에 전역으로 사용하지 않고 3번째 for문 내에서 선언해서 사용해주세요.
- 그리고 반복횟수도 i 는 cardNum - 2, j는 cardNum - 1 만큼만 반복하시는게 더 연산횟수 최적화가 가능합니다.
- 아래 코드는
```js
if (tempSum > result && tempSum <= dealNum) {
  result = tempSum;
}
```
- Math.max를 이용해서 아래와 같이 풀이도 가능합니다.
```js
if (tempSum <= dealNum) {
  result = Math.max(result, tempSum);
}
```
```

Reply:
오오 이렇게 함수로도 만들고 사용할 수 잇군요.....!! 하나배웠습니다!!!!! 👍🏻

Reply:
오 Math.max로 저렇게 하는게 훨씬 깔끔하네요 명확하고! 피드백 참고하겠습니다!